uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Service/LeaveRequest.php

```diff
@@ -59,6 +66,8 @@ public function __construct(
     $this->leaveBalanceChangeService = $leaveBalanceChangeService;
     $this->leaveRequestStatusMatrixService = $leaveRequestStatusMatrixService;
     $this->leaveRequestRightsService = $leaveRequestRightsService;
+    $publicHolidayLeaveRequestDeletionFactory = new PublicHolidayLeaveRequestDeletionFactory();
```

Comment:
I just noticed this now, but if the only method called is a static one, why are you creating a new instance here?

Reply:
Nice catch!